vm/src/builtins/object.rs

```diff
@@ -184,8 +185,25 @@ impl PyBaseObject {
 
     /// Return repr(self).
     #[pymethod(magic)]
-    fn repr(zelf: PyObjectRef) -> String {
-        format!("<{} object at {:#x}>", zelf.class().name, zelf.get_id())
+    fn repr(zelf: PyObjectRef, vm: &VirtualMachine) -> String {
+        let class = zelf.class();
+        let module = class.module(vm);
+        let module = module.downcast_ref::<PyStr>().map(|m| m.as_str());
+        match module {
```

Comment:
You can just combine these lines instead of re-binding to `module`:

```suggestion
        match class.module(vm).downcast_ref::<PyStr>().map(|m| m.as_str()) {
```